### PR TITLE
Move binary publish from bintray to GCP

### LINKED ISSUE #4183: Move published artifacts to google storage
User statement:  As an engineer and customer of VIC, I want the builds artifacts to be published in an accessible and supported location

Details: We have decided to consolidate our image and binary storage locations and google has been chosen. We need to move the publishing over to the new location and orchestrate all the documentation and nightly testing and any other scripts that are pointing to bintray need to be moved in relatively close order.  The change should be fairly easy using this guide:
https://github.com/drone-plugins/drone-google-cloudstorage/blob/master/DOCS.md

Acceptance criteria: Build artifacts and documentation all point to google cloud storage.

### PR DESCRIPTION
**Pull request by mhagen-vmware**
fixes #4183 

Changes binary publish to google storage, cleans up some tests that download builds, and adds new publish functionality so that we don't have to manually copy binaries anymore to our release bucket on tag event.  Had to touch a lot of areas including the OVA build process as well.

### CODE REVIEW

File: tests/manual-test-cases/Group16-WindowsClient/16-1-Windows-VIC-Machine.robot [hunk 1]
```diff
@@ -18,7 +18,7 @@ Resource  ../../resources/Util.robot
 Test Setup  Set Test Environment Variables
 
 *** Variables ***
-${ver}  0.6.0
+${ver}  8351
```
Comment (andrewtchin) [inline tests/manual-test-cases/Group16-WindowsClient/16-1-Windows-VIC-Machine.robot]: What does this refer to?

Comment (andrewtchin) [inline tests/manual-test-cases/Group16-WindowsClient/16-1-Windows-VIC-Machine.robot]: Oh build number

Comment (mhagen-vmware) [inline tests/manual-test-cases/Group16-WindowsClient/16-1-Windows-VIC-Machine.robot]: just a severely outdated test that needs love in a different PR.  I used the most recent one that we have in another test, I believe 8351 was 0.8 ultimately.

Comment (andrewtchin) [inline tests/manual-test-cases/Group16-WindowsClient/16-1-Windows-VIC-Machine.robot]: Are you going to push this build to the bucket?

Comment (mhagen-vmware) [inline tests/manual-test-cases/Group16-WindowsClient/16-1-Windows-VIC-Machine.robot]: already done

Comment (andrewtchin) [inline tests/manual-test-cases/Group16-WindowsClient/16-1-Windows-VIC-Machine.robot]: Oh I was looking at releases bucket

**Comment by rajanashok** [comment]
Nice! lgtm

**Pull request merged**